希望重新出一个config.json配置教程 现在的配置已经不懂了

好久没更新了 最近更新了一下 遇到一堆问题
还有 请问 现在的DDNS 配置 支持2个供应商 共同解析吗？就是`dnspod_com`+`cloudflare`
如果没有 希望支持~仅仅在1个配置文件中 更新多个服务商
还有关于打乱配置文件中的顺序 是否会出错？ 比如"id"下面跟“token”
试过很多次配置 均无效 才来反馈 真的太难了！

DDNS版本：DDNS-2.7.2
系统版本：win10

`config.json`配置文件
```javascript
{
  "$schema": "https://ddns.newfuture.cc/schema/v2.json",
  "debug": true,
  "dns": "dnspod_com",
  "id": "****",
  "index4": "default",
  "index6": "default",
  "ipv4": "lm.lorzl.ml",
  "ipv6": "lm.lorzl.ml",
  "proxy": null,
  "token": "*******"
}
```

运行`ddns.exe`不会获取公网IPV4地址

```console
C:\Users\Administrator>E:\caddy\DDNS-2.7.2\ddns.exe

newfuture.cc(A) ==> 6.6.6.1 [via DIRECT]
ERROR:root:invalid domain: [ newfuture.cc ]

ddns.newfuture.cc(A) ==> 6.6.6.1 [via DIRECT]
ERROR:root:invalid domain: [ ddns.newfuture.cc ]

newfuture.cc(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
ERROR:root:invalid domain: [ newfuture.cc ]

ipv6.ddns.newfuture.cc(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
ERROR:root:invalid domain: [ ipv6.ddns.newfuture.cc ]
```

运行`run.bat`就比较恐怖了 一大串 可以说windows支持基本就exe了……？
```console
2019-10-31 22:50:31,700 <run.main> 164@E:\caddy\DDNS-2.7.2\run.py
[INFO] DDNS[${BUILD_SOURCEBRANCHNAME}@${BUILD_DATE}] run: nt,win32
2019-10-31 22:50:31,701 <cache.load> 48@E:\caddy\DDNS-2.7.2\util\cache.py
[DEBUG] load cache data from C:\Users\Administrator\AppData\Local\Temp\ddns.cache

l(A) ==> 6.6.6.1 [via DIRECT]
2019-10-31 22:50:31,709 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>l(A)
2019-10-31 22:50:31,709 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:31,710 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ l ]

m(A) ==> 6.6.6.1 [via DIRECT]
2019-10-31 22:50:31,710 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>m(A)
2019-10-31 22:50:31,710 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:31,711 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ m ]

.(A) ==> 6.6.6.1 [via DIRECT]
2019-10-31 22:50:31,711 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>.(A)
2019-10-31 22:50:31,711 <dnspod.request> 45@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'domain': '.', 'format': 'json', 'user_token': '***'}
2019-10-31 22:50:32,361 <dnspod.request> 65@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[DEBUG] Domain.Info : result:{'status': {'code': '-1', 'message': 'Login fail, please check login info', 'created_at': '2019-10-31 14:50:03'}}
2019-10-31 22:50:32,363 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:32,368 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ . ]

l(A) ==> 6.6.6.1 [via DIRECT]
2019-10-31 22:50:32,373 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>l(A)
2019-10-31 22:50:32,376 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:32,378 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ l ]

o(A) ==> 6.6.6.1 [via DIRECT]
2019-10-31 22:50:32,384 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>o(A)
2019-10-31 22:50:32,386 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:32,389 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ o ]

r(A) ==> 6.6.6.1 [via DIRECT]
2019-10-31 22:50:32,395 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>r(A)
2019-10-31 22:50:32,398 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:32,400 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ r ]

z(A) ==> 6.6.6.1 [via DIRECT]
2019-10-31 22:50:32,406 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>z(A)
2019-10-31 22:50:32,408 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:32,411 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ z ]

l(A) ==> 6.6.6.1 [via DIRECT]
2019-10-31 22:50:32,416 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>l(A)
2019-10-31 22:50:32,418 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:32,420 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ l ]

.(A) ==> 6.6.6.1 [via DIRECT]
2019-10-31 22:50:32,426 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>.(A)
2019-10-31 22:50:32,428 <dnspod.request> 45@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'domain': '.', 'format': 'json', 'user_token': '***'}
2019-10-31 22:50:32,925 <dnspod.request> 65@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[DEBUG] Domain.Info : result:{'status': {'code': '-1', 'message': 'Login fail, please check login info', 'created_at': '2019-10-31 14:50:03'}}
2019-10-31 22:50:32,927 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:32,929 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ . ]

m(A) ==> 6.6.6.1 [via DIRECT]
2019-10-31 22:50:32,934 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>m(A)
2019-10-31 22:50:32,936 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:32,937 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ m ]

l(A) ==> 6.6.6.1 [via DIRECT]
2019-10-31 22:50:32,942 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>l(A)
2019-10-31 22:50:32,944 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:32,947 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ l ]

l(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
2019-10-31 22:50:32,951 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>l(AAAA)
2019-10-31 22:50:32,954 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:32,955 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ l ]

m(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
2019-10-31 22:50:32,961 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>m(AAAA)
2019-10-31 22:50:32,963 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:32,965 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ m ]

.(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
2019-10-31 22:50:32,969 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>.(AAAA)
2019-10-31 22:50:32,972 <dnspod.request> 45@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'domain': '.', 'format': 'json', 'user_token': '***'}
2019-10-31 22:50:33,423 <dnspod.request> 65@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[DEBUG] Domain.Info : result:{'status': {'code': '-1', 'message': 'Login fail, please check login info', 'created_at': '2019-10-31 14:50:04'}}
2019-10-31 22:50:33,425 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:33,427 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ . ]

l(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
2019-10-31 22:50:33,432 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>l(AAAA)
2019-10-31 22:50:33,435 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:33,437 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ l ]

o(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
2019-10-31 22:50:33,441 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>o(AAAA)
2019-10-31 22:50:33,444 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:33,446 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ o ]

r(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
2019-10-31 22:50:33,449 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>r(AAAA)
2019-10-31 22:50:33,452 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:33,454 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ r ]

z(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
2019-10-31 22:50:33,458 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>z(AAAA)
2019-10-31 22:50:33,460 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:33,462 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ z ]

l(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
2019-10-31 22:50:33,467 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>l(AAAA)
2019-10-31 22:50:33,469 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:33,471 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ l ]

.(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
2019-10-31 22:50:33,474 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>.(AAAA)
2019-10-31 22:50:33,477 <dnspod.request> 45@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'domain': '.', 'format': 'json', 'user_token': '***'}
2019-10-31 22:50:33,971 <dnspod.request> 65@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[DEBUG] Domain.Info : result:{'status': {'code': '-1', 'message': 'Login fail, please check login info', 'created_at': '2019-10-31 14:50:04'}}
2019-10-31 22:50:33,974 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:33,976 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ . ]

m(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
2019-10-31 22:50:33,982 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>m(AAAA)
2019-10-31 22:50:33,984 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:33,987 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ m ]

l(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
2019-10-31 22:50:33,991 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>l(AAAA)
2019-10-31 22:50:33,994 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-10-31 22:50:33,997 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ l ]
请按任意键继续. . .
```

```console
2019-11-01 01:21:11,110 <run.main> 164@E:\caddy\DDNS-2.7.2\run.py
[INFO] DDNS[${BUILD_SOURCEBRANCHNAME}@${BUILD_DATE}] run: nt,win32
2019-11-01 01:21:11,111 <cache.load> 48@E:\caddy\DDNS-2.7.2\util\cache.py
[DEBUG] load cache data from C:\Users\Administrator\AppData\Local\Temp\ddns.cache
========================= Fri Nov  1 01:21:11 2019 =========================
2019-11-01 01:21:11,112 <ip._open> 51@E:\caddy\DDNS-2.7.2\util\ip.py
[DEBUG] open: https://api-ipv4.ip.sb/ip
2019-11-01 01:21:11,605 <ip._open> 55@E:\caddy\DDNS-2.7.2\util\ip.py
[DEBUG] response: 218.5.40.11


lm.lorzl.ml(A) ==> 218.5.40.11 [via DIRECT]
2019-11-01 01:21:11,607 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>lm.lorzl.ml(A)
2019-11-01 01:21:11,607 <dnspod.request> 45@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'format': 'json', 'domain': 'lorzl.ml', 'user_token': '***'}
2019-11-01 01:21:12,093 <dnspod.request> 65@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[DEBUG] Domain.Info : result:{'status': {'created_at': '2019-10-31 17:20:42', 'message': 'Login fail, please check login info', 'code': '-1'}}
2019-11-01 01:21:12,094 <dnspod.request> 45@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'format': 'json', 'domain': 'lm.lorzl.ml', 'user_token': '***'}
2019-11-01 01:21:12,572 <dnspod.request> 65@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[DEBUG] Domain.Info : result:{'status': {'created_at': '2019-10-31 17:20:43', 'message': 'Login fail, please check login info', 'code': '-1'}}
2019-11-01 01:21:12,574 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-11-01 01:21:12,574 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ lm.lorzl.ml ]
2019-11-01 01:21:12,574 <ip._open> 51@E:\caddy\DDNS-2.7.2\util\ip.py
[DEBUG] open: https://api-ipv6.ip.sb/ip
2019-11-01 01:21:15,007 <ip._open> 55@E:\caddy\DDNS-2.7.2\util\ip.py
[DEBUG] response: 240e:d0:3554:d700:5e3:6078:ca9:e8e1


lm.lorzl.ml(AAAA) ==> 240e:d0:3554:d700:5e3:6078:ca9:e8e1 [via DIRECT]
2019-11-01 01:21:15,014 <dnspod.update_record> 148@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] >>>>>lm.lorzl.ml(AAAA)
2019-11-01 01:21:15,017 <dnspod.request> 45@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'format': 'json', 'domain': 'lorzl.ml', 'user_token': '***'}
2019-11-01 01:21:15,499 <dnspod.request> 65@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[DEBUG] Domain.Info : result:{'status': {'created_at': '2019-10-31 17:20:46', 'message': 'Login fail, please check login info', 'code': '-1'}}
2019-11-01 01:21:15,500 <dnspod.request> 45@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'format': 'json', 'domain': 'lm.lorzl.ml', 'user_token': '***'}
2019-11-01 01:21:15,963 <dnspod.request> 65@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[DEBUG] Domain.Info : result:{'status': {'created_at': '2019-10-31 17:20:46', 'message': 'Login fail, please check login info', 'code': '-1'}}
2019-11-01 01:21:15,968 <dnspod.get_domain_info> 88@E:\caddy\DDNS-2.7.2\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-11-01 01:21:15,973 <run.change_dns_record> 116@E:\caddy\DDNS-2.7.2\run.py
[ERROR] invalid domain: [ lm.lorzl.ml ]
2019-11-01 01:21:15,979 <cache.sync> 85@E:\caddy\DDNS-2.7.2\util\cache.py
[DEBUG] save cache data to C:\Users\Administrator\AppData\Local\Temp\ddns.cache
请按任意键继续. . .
```

`配置模式1`
```python
{
	"$schema": "https://ddns.newfuture.cc/schema/v2.json",
	"id": "****",
	"token": "********",
	"dns": "dnspod_com",
	"ipv4": ["lm.lorzl.ml"],
	"ipv6": ["lm.lorzl.ml"],
	"index4": "public",
	"index6": "public",
	"proxy": "DIRECT",
	"debug": false 
}
```

```console
ERROR:root: Config file `E:\caddy\DDNS-2.7.2\config.json` does not exist!
New template configure file `E:\caddy\DDNS-2.7.2\config.json` is generated.
请按任意键继续. . .
```

`配置模式2`=开启调试模式

```python
{
	"$schema": "https://ddns.newfuture.cc/schema/v2.json",
	"debug": true,
	"dns": "dnspod_com",
	"id": "****",
	"index4": "default",
	"index6": "default",
	"ipv4": ["lm.lorzl.ml"],
	"ipv6": ["lm.lorzl.ml"],
  "proxy": null,
  "token": "*****************"
}
```

### 错误日志

```console
2019-11-02 02:32:09,589 <run.main> 165@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\run.py
[INFO] DDNS[${BUILD_SOURCEBRANCHNAME}@${BUILD_DATE}] run: nt,win32
2019-11-02 02:32:09,589 <cache.load> 48@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\util\cache.py
[DEBUG] load cache data from C:\Users\Administrator\AppData\Local\Temp\ddns.cache
========================= Sat Nov  2 02:32:09 2019 =========================

lm.lorzl.ml(A) ==> 6.6.6.1 [via DIRECT]
2019-11-02 02:32:09,597 <dnspod.update_record> 148@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py 
[INFO] >>>>>lm.lorzl.ml(A)
2019-11-02 02:32:09,597 <dnspod.request> 46@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'domain': 'lorzl.ml', 'user_token': '***', 'format': 'json'}
2019-11-02 02:32:10,340 <dnspod.request> 61@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[WARNING] Domain.Info : error:b''
2019-11-02 02:32:10,341 <dnspod.request> 46@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'domain': 'lm.lorzl.ml', 'user_token': '***', 'format': 'json'}
2019-11-02 02:32:10,919 <dnspod.request> 61@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[WARNING] Domain.Info : error:b''
2019-11-02 02:32:10,919 <dnspod.get_domain_info> 88@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-11-02 02:32:10,920 <run.change_dns_record> 117@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\run.py
[ERROR] invalid domain: [ lm.lorzl.ml ]

lm.lorzl.ml(AAAA) ==> 240e:d0:3554:d700:bc:94cf:375d:62e9 [via DIRECT]
2019-11-02 02:32:10,921 <dnspod.update_record> 148@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py 
[INFO] >>>>>lm.lorzl.ml(AAAA)
2019-11-02 02:32:10,921 <dnspod.request> 46@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'domain': 'lorzl.ml', 'user_token': '***', 'format': 'json'}
2019-11-02 02:32:11,505 <dnspod.request> 61@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[WARNING] Domain.Info : error:b''
2019-11-02 02:32:11,506 <dnspod.request> 46@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'domain': 'lm.lorzl.ml', 'user_token': '***', 'format': 'json'}
2019-11-02 02:32:12,075 <dnspod.request> 61@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[WARNING] Domain.Info : error:b''
2019-11-02 02:32:12,076 <dnspod.get_domain_info> 88@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-11-02 02:32:12,078 <run.change_dns_record> 117@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\run.py
[ERROR] invalid domain: [ lm.lorzl.ml ]
2019-11-02 02:32:12,081 <cache.sync> 85@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\util\cache.py
[DEBUG] save cache data to C:\Users\Administrator\AppData\Local\Temp\ddns.cache
请按任意键继续. . .
```

`配置1`
```python
{
  "$schema": "https://ddns.newfuture.cc/schema/v2.json",
  "id": "****",
  "token": "*************",
  "dns": "dnspod_com",
  "ipv4": ["lm.lorzl.ml"],
  "ipv6": ["lm.lorzl.ml"],
  "index4": 0,
  "index6": "public",
  "proxy": "DIRECT",
  "debug": false
}
```
设置`"proxy": "DIRECT"`好像他获取到的是我的内网 改成null也是一样
```console
WARNING:root:Cache file is out of dated.

lm.lorzl.ml(A) ==> 6.6.6.1 [via DIRECT]
WARNING:root:Domain.Info : error[302]:
WARNING:root:Domain.Info : error[302]:
ERROR:root:invalid domain: [ lm.lorzl.ml ]

lm.lorzl.ml(AAAA) ==> 240e:d0:3554:d700:7d48:662:8504:75ee [via DIRECT]
WARNING:root:Domain.Info : error[302]:
WARNING:root:Domain.Info : error[302]:
ERROR:root:invalid domain: [ lm.lorzl.ml ]
请按任意键继续. . .

```
开启`"debug":true`
```console
DDNS[ ${BUILD_SOURCEBRANCHNAME}@${BUILD_DATE} ] run: nt win32
Configuration was loaded from <== C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\config.json
========================= Sat Nov  2 15:18:37 2019 =========================
2019-11-02 15:18:37,029 <cache.load> 48@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\util\cache.py
[DEBUG] load cache data from C:\Users\Administrator\AppData\Local\Temp\ddns.cache
2019-11-02 15:18:37,030 <run.main> 177@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\run.py
[WARNING] Cache file is out of dated.
2019-11-02 15:18:37,050 <ip.local_v4> 45@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\util\ip.py
[DEBUG] [(<AddressFamily.AF_INET: 2>, 0, 0, '', ('6.6.6.1', 0)), (<AddressFamily.AF_INET: 2>, 0, 0, '', ('6.6.6.1', 0))]

lm.lorzl.ml(A) ==> 6.6.6.1 [via DIRECT]
2019-11-02 15:18:37,050 <dnspod.update_record> 148@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py 
[INFO] >>>>>lm.lorzl.ml(A)
2019-11-02 15:18:37,051 <dnspod.request> 46@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'user_token': '***', 'format': 'json', 'domain': 'lorzl.ml'}
2019-11-02 15:18:37,859 <dnspod.request> 61@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[WARNING] Domain.Info : error[302]:
2019-11-02 15:18:37,860 <dnspod.request> 46@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'user_token': '***', 'format': 'json', 'domain': 'lm.lorzl.ml'}
2019-11-02 15:18:39,498 <dnspod.request> 61@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[WARNING] Domain.Info : error[302]:
2019-11-02 15:18:39,498 <dnspod.get_domain_info> 88@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-11-02 15:18:39,499 <run.change_dns_record> 117@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\run.py
[ERROR] invalid domain: [ lm.lorzl.ml ]
2019-11-02 15:18:39,499 <ip._open> 51@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\util\ip.py
[DEBUG] open: https://api-ipv6.ip.sb/ip
2019-11-02 15:18:42,816 <ip._open> 55@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\util\ip.py
[DEBUG] response: 240e:d0:3554:d700:7d48:662:8504:75ee


lm.lorzl.ml(AAAA) ==> 240e:d0:3554:d700:7d48:662:8504:75ee [via DIRECT]
2019-11-02 15:18:42,821 <dnspod.update_record> 148@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py 
[INFO] >>>>>lm.lorzl.ml(AAAA)
2019-11-02 15:18:42,821 <dnspod.request> 46@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'user_token': '***', 'format': 'json', 'domain': 'lorzl.ml'}
2019-11-02 15:18:43,425 <dnspod.request> 61@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[WARNING] Domain.Info : error[302]:
2019-11-02 15:18:43,426 <dnspod.request> 46@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'user_token': '***', 'format': 'json', 'domain': 'lm.lorzl.ml'}
2019-11-02 15:18:44,027 <dnspod.request> 61@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[WARNING] Domain.Info : error[302]:
2019-11-02 15:18:44,028 <dnspod.get_domain_info> 88@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-11-02 15:18:44,029 <run.change_dns_record> 117@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\run.py
[ERROR] invalid domain: [ lm.lorzl.ml ]
2019-11-02 15:18:44,030 <cache.sync> 85@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\util\cache.py
[DEBUG] save cache data to C:\Users\Administrator\AppData\Local\Temp\ddns.cache
请按任意键继续. . .
```


`配置2`
```python
{
  "$schema": "https://ddns.newfuture.cc/schema/v2.json",
  "debug": true,
  "dns": "dnspod_com",
  "id": "*****",
  "index4": "default",
  "index6": "default",
  "ipv4": [
    "lm.lorzl.ml"
  ],
  "ipv6": [
    "lm.lorzl.ml"
  ],
  "proxy": null,
  "token": "************"
}
```
不开BUG调试错误和配置1相同，以下为BUG日志
```console
DDNS[ ${BUILD_SOURCEBRANCHNAME}@${BUILD_DATE} ] run: nt win32
Configuration was loaded from <== C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\config.json
========================= Sat Nov  2 15:28:19 2019 =========================
2019-11-02 15:28:19,406 <cache.load> 48@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\util\cache.py
[DEBUG] load cache data from C:\Users\Administrator\AppData\Local\Temp\ddns.cache
2019-11-02 15:28:19,407 <run.main> 177@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\run.py
[WARNING] Cache file is out of dated.

lm.lorzl.ml(A) ==> 6.6.6.1 [via DIRECT]
2019-11-02 15:28:19,414 <dnspod.update_record> 148@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py 
[INFO] >>>>>lm.lorzl.ml(A)
2019-11-02 15:28:19,414 <dnspod.request> 46@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'domain': 'lorzl.ml', 'user_token': '***', 'format': 'json'}
2019-11-02 15:28:20,169 <dnspod.request> 61@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[WARNING] Domain.Info : error[302]:
2019-11-02 15:28:20,169 <dnspod.request> 46@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'domain': 'lm.lorzl.ml', 'user_token': '***', 'format': 'json'}
2019-11-02 15:28:20,737 <dnspod.request> 61@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[WARNING] Domain.Info : error[302]:
2019-11-02 15:28:20,738 <dnspod.get_domain_info> 88@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-11-02 15:28:20,739 <run.change_dns_record> 117@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\run.py
[ERROR] invalid domain: [ lm.lorzl.ml ]

lm.lorzl.ml(AAAA) ==> 240e:d0:3554:d700:7d48:662:8504:75ee [via DIRECT]
2019-11-02 15:28:20,746 <dnspod.update_record> 148@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py 
[INFO] >>>>>lm.lorzl.ml(AAAA)
2019-11-02 15:28:20,751 <dnspod.request> 46@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'domain': 'lorzl.ml', 'user_token': '***', 'format': 'json'}
2019-11-02 15:28:21,355 <dnspod.request> 61@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[WARNING] Domain.Info : error[302]:
2019-11-02 15:28:21,355 <dnspod.request> 46@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] api.dnspod.com/Domain.Info : {'domain': 'lm.lorzl.ml', 'user_token': '***', 'format': 'json'}
2019-11-02 15:28:21,923 <dnspod.request> 61@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[WARNING] Domain.Info : error[302]:
2019-11-02 15:28:21,924 <dnspod.get_domain_info> 88@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\dns\dnspod.py
[INFO] domain_id: None, sub: None
2019-11-02 15:28:21,926 <run.change_dns_record> 117@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\run.py
[ERROR] invalid domain: [ lm.lorzl.ml ]
2019-11-02 15:28:21,928 <cache.sync> 85@C:\Users\Administrator\Desktop\DDNS-fix-dnspod_com-104\util\cache.py
[DEBUG] save cache data to C:\Users\Administrator\AppData\Local\Temp\ddns.cache
请按任意键继续. . .

```

--- a/dns/alidns.py
+++ b/dns/alidns.py
@@ -82,14 +82,14 @@ def request(param=None, **params):
     conn.request(API.METHOD, '/', urlencode(params),
                  {"Content-type": "application/x-www-form-urlencoded"})
     response = conn.getresponse()
-    data = response.read()
+    data = response.read().decode('utf8')
     conn.close()
 
     if response.status < 200 or response.status >= 300:
-        warning('%s : error:%s', params['Action'], data)
+        warning('%s : error[%d]: %s', params['Action'], response.status, data)
         raise Exception(data)
     else:
-        data = jsondecode(data.decode('utf8'))
+        data = jsondecode(data)
         debug('%s : result:%s', params['Action'], data)
         return data
 

--- a/dns/alidns.py
+++ b/dns/alidns.py
@@ -26,9 +26,12 @@ except ImportError:
 __author__ = 'New Future'
 # __all__ = ["request", "ID", "TOKEN", "PROXY"]
 
-ID = "id"
-TOKEN = "TOKEN"
-PROXY = None  # 代理设置
+
+class Config:
+    ID = "id"
+    TOKEN = "TOKEN"
+    PROXY = None  # 代理设置
+    TTL = 600
 
 
 class API:
@@ -44,7 +47,7 @@ def signature(params):
     params.update({
         'Format': 'json',
         'Version': '2015-01-09',
-        'AccessKeyId': ID,
+        'AccessKeyId': Config.ID,
         'Timestamp': datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%SZ'),
         'SignatureMethod': 'HMAC-SHA1',
         'SignatureNonce': uuid4(),
@@ -55,7 +58,7 @@ def signature(params):
     sign = API.METHOD + "&" + quote_plus("/") + "&" + quote(query, safe='')
     debug("signString: %s", sign)
 
-    sign = hmac((TOKEN + "&").encode('utf-8'),
+    sign = hmac((Config.TOKEN + "&").encode('utf-8'),
                 sign.encode('utf-8'), sha1).digest()
     sign = b64encode(sign).strip()
     params["Signature"] = sign
@@ -71,8 +74,8 @@ def request(param=None, **params):
     params = signature(params)
     info("%s: %s", API.SITE, params)
 
-    if PROXY:
-        conn = HTTPSConnection(PROXY)
+    if Config.PROXY:
+        conn = HTTPSConnection(Config.PROXY)
         conn.set_tunnel(API.SITE, 443)
     else:
         conn = HTTPSConnection(API.SITE)

--- a/dns/cloudflare.py
+++ b/dns/cloudflare.py
@@ -20,9 +20,12 @@ except ImportError:
 
 __author__ = 'TongYifan'
 
-ID = "AUTH EMAIL"  # CloudFlare 验证的是用户Email，等同于其他平台的userID
-TOKEN = "API KEY"
-PROXY = None  # 代理设置
+
+class Config:
+    ID = "AUTH EMAIL"  # CloudFlare 验证的是用户Email，等同于其他平台的userID
+    TOKEN = "API KEY"
+    PROXY = None  # 代理设置
+    TTL = 600
 
 
 class API:
@@ -38,8 +41,8 @@ def request(method, action, param=None, **params):
         params.update(param)
 
     info("%s/%s : %s", API.SITE, action, params)
-    if PROXY:
-        conn = HTTPSConnection(PROXY)
+    if Config.PROXY:
+        conn = HTTPSConnection(Config.PROXY)
         conn.set_tunnel(API.SITE, 443)
     else:
         conn = HTTPSConnection(API.SITE)
@@ -54,8 +57,8 @@ def request(method, action, param=None, **params):
         params = None
     conn.request(method, '/client/v4/zones' + action, params,
                  {"Content-type": "application/json",
-                  "X-Auth-Email": ID,
-                  "X-Auth-Key": TOKEN})
+                  "X-Auth-Email": Config.ID,
+                  "X-Auth-Key": Config.TOKEN})
     response = conn.getresponse()
     res = response.read()
     conn.close()

--- a/dns/cloudflare.py
+++ b/dns/cloudflare.py
@@ -7,7 +7,7 @@ https://api.cloudflare.com/#dns-records-for-a-zone-properties
 """
 
 from json import loads as jsondecode, dumps as jsonencode
-from logging import debug, info
+from logging import debug, info, warning
 
 try:
     # python 2
@@ -60,12 +60,13 @@ def request(method, action, param=None, **params):
                   "X-Auth-Email": Config.ID,
                   "X-Auth-Key": Config.TOKEN})
     response = conn.getresponse()
-    res = response.read()
+    res = response.read().decode('utf8')
     conn.close()
     if response.status < 200 or response.status >= 300:
+        warning('%s : error[%d]:%s', action, response.status, res)
         raise Exception(res)
     else:
-        data = jsondecode(res.decode('utf8'))
+        data = jsondecode(res)
         debug('%s : result:%s', action, data)
         if not data:
             raise Exception("Empty Response")

--- a/dns/dnscom.py
+++ b/dns/dnscom.py
@@ -26,9 +26,12 @@ except ImportError:
 __author__ = 'Bigjin'
 # __all__ = ["request", "ID", "TOKEN", "PROXY"]
 
-ID = "id"
-TOKEN = "TOKEN"
-PROXY = None  # 代理设置
+
+class Config:
+    ID = "id"
+    TOKEN = "TOKEN"
+    PROXY = None  # 代理设置
+    TTL = 600
 
 
 class API:
@@ -42,7 +45,7 @@ def signature(params):
     计算签名,返回签名后的查询参数
     """
     params.update({
-        'apiKey': ID,
+        'apiKey': Config.ID,
         'timestamp': mktime(datetime.now().timetuple()),
     })
     query = urlencode(sorted(params.items()))
@@ -50,7 +53,7 @@ def signature(params):
     sign = query
     debug("signString: %s", sign)
 
-    sign = md5((sign + TOKEN).encode('utf-8')).hexdigest()
+    sign = md5((sign + Config.TOKEN).encode('utf-8')).hexdigest()
     params["hash"] = sign
 
     return params
@@ -65,8 +68,8 @@ def request(action, param=None, **params):
     params = signature(params)
     info("%s/api/%s/ : params:%s", API.SITE, action, params)
 
-    if PROXY:
-        conn = HTTPSConnection(PROXY)
+    if Config.PROXY:
+        conn = HTTPSConnection(Config.PROXY)
         conn.set_tunnel(API.SITE, 443)
     else:
         conn = HTTPSConnection(API.SITE)

--- a/dns/dnscom.py
+++ b/dns/dnscom.py
@@ -9,7 +9,7 @@ http://open.dns.com/
 
 from hashlib import md5
 from json import loads as jsondecode
-from logging import debug, info
+from logging import debug, info, warning
 from time import mktime
 from datetime import datetime
 
@@ -77,13 +77,14 @@ def request(action, param=None, **params):
     conn.request(API.METHOD, '/api/' + action + '/', urlencode(params),
                  {"Content-type": "application/x-www-form-urlencoded"})
     response = conn.getresponse()
-    result = response.read()
+    result = response.read().decode('utf8')
     conn.close()
 
     if response.status < 200 or response.status >= 300:
+        warning('%s : error[%d]:%s', action, response.status, result)        
         raise Exception(result)
     else:
-        data = jsondecode(result.decode('utf8'))
+        data = jsondecode(result)
         debug('%s : result:%s', action, data)
         if data.get('code') != 0:
             raise Exception("api error:", data.get('message'))

--- a/dns/dnspod.py
+++ b/dns/dnspod.py
@@ -8,6 +8,7 @@ http://www.dnspod.cn/docs/domains.html
 
 from json import loads as jsondecode
 from logging import debug, info, warning
+from os import environ
 try:
     # python 2
     from httplib import HTTPSConnection
@@ -51,8 +52,10 @@ def request(action, param=None, **params):
     else:
         conn = HTTPSConnection(API.SITE)
 
-    conn.request(API.METHOD, '/' + action, urlencode(params),
-                 {"Content-type": "application/x-www-form-urlencoded"})
+    conn.request(API.METHOD, '/' + action, urlencode(params), {
+        "Content-type": "application/x-www-form-urlencoded",
+        "User-Agent": "DDNS/%s (ddns@newfuture.cc)" % environ.get("DDNS_VERSION", "1.0.0")
+    })
     response = conn.getresponse()
     res = response.read().decode('utf8')
     conn.close()

--- a/dns/dnspod.py
+++ b/dns/dnspod.py
@@ -20,10 +20,11 @@ except ImportError:
 __author__ = 'New Future'
 
 
-ID = "token id"
-TOKEN = "token key"
-PROXY = None  # 代理设置
-TTL = 600
+class Config:
+    ID = "token id"
+    TOKEN = "token key"
+    PROXY = None  # 代理设置
+    TTL = 600
 
 
 class API:
@@ -43,10 +44,9 @@ def request(action, param=None, **params):
 
     params.update({API.TOKEN_PARAM: '***', 'format': 'json'})
     info("%s/%s : %s", API.SITE, action, params)
-    params[API.TOKEN_PARAM] = "%s,%s" % (ID, TOKEN)
-
-    if PROXY:
-        conn = HTTPSConnection(PROXY)
+    params[API.TOKEN_PARAM] = "%s,%s" % (Config.ID, Config.TOKEN)
+    if Config.PROXY:
+        conn = HTTPSConnection(Config.PROXY)
         conn.set_tunnel(API.SITE, 443)
     else:
         conn = HTTPSConnection(API.SITE)
@@ -169,7 +169,7 @@ def update_record(domain, value, record_type="A"):
     else:  # create
         # http://www.dnspod.cn/docs/records.html#record-create
         res = request("Record.Create", domain_id=domainid, value=value,
-                      sub_domain=sub, record_type=record_type, record_line=API.DEFAULT, ttl=TTL)
+                      sub_domain=sub, record_type=record_type, record_line=API.DEFAULT, ttl=Config.TTL)
         if res:
             did = res.get("record")["id"]
             get_records.records[domainid][did] = res.get("record")

--- a/dns/dnspod.py
+++ b/dns/dnspod.py
@@ -54,14 +54,14 @@ def request(action, param=None, **params):
     conn.request(API.METHOD, '/' + action, urlencode(params),
                  {"Content-type": "application/x-www-form-urlencoded"})
     response = conn.getresponse()
-    res = response.read()
+    res = response.read().decode('utf8')
     conn.close()
 
     if response.status < 200 or response.status >= 300:
-        warning('%s : error:%s', action, res)
+        warning('%s : error[%d]:%s', action, response.status, res)
         raise Exception(res)
     else:
-        data = jsondecode(res.decode('utf8'))
+        data = jsondecode(res)
         debug('%s : result:%s', action, data)
         if not data:
             raise Exception("empty response")

--- a/run.py
+++ b/run.py
@@ -85,11 +85,12 @@ def get_ip(ip_type):
         return False
     elif str(index).isdigit():  # 数字 local eth
         value = getattr(ip, "local_v" + ip_type)(index)
-    elif index.startswith('cmd:'): # cmd
+    elif index.startswith('cmd:'):  # cmd
         value = str(check_output(index[4:]).strip().decode('utf-8'))
-    elif index.startswith('shell:'): # shell
-        value = str(check_output(index[6:], shell=True).strip().decode('utf-8'))
-    elif index.startswith('url:'): # 自定义 url
+    elif index.startswith('shell:'):  # shell
+        value = str(check_output(
+            index[6:], shell=True).strip().decode('utf-8'))
+    elif index.startswith('url:'):  # 自定义 url
         value = getattr(ip, "public_v" + ip_type)(index[4:])
     elif index.startswith('regex:'):  # 正则 regex
         value = getattr(ip, "regex_v" + ip_type)(index[6:])
@@ -155,7 +156,7 @@ def main():
     # Dynamicly import the dns module as configuration
     dns_provider = str(get_config('dns', 'dnspod').lower())
     dns = getattr(__import__('dns', fromlist=[dns_provider]), dns_provider)
-    dns.ID, dns.TOKEN = get_config('id'), get_config('token')
+    dns.Config.ID, dns.Config.TOKEN = get_config('id'), get_config('token')
     if get_config('debug'):
         ip.DEBUG = get_config('debug')
         basicConfig(

--- a/run.py
+++ b/run.py
@@ -152,7 +152,8 @@ def main():
                         action='version', version=__version__)
     parser.add_argument('-c', '--config',
                         default="config.json", help="run with config file [配置文件路径]")
-    get_config(path=parser.parse_args().config)
+    config_file = parser.parse_args().config
+    get_config(path=config_file)
     # Dynamicly import the dns module as configuration
     dns_provider = str(get_config('dns', 'dnspod').lower())
     dns = getattr(__import__('dns', fromlist=[dns_provider]), dns_provider)
@@ -162,7 +163,8 @@ def main():
         basicConfig(
             level=DEBUG,
             format='%(asctime)s <%(module)s.%(funcName)s> %(lineno)d@%(pathname)s \n[%(levelname)s] %(message)s')
-        info("DDNS[%s] run: %s,%s", __version__, os_name, sys.platform)
+        print("DDNS[", __version__, "] run:", os_name, sys.platform)
+        print("Configuration was loaded from <==", path.abspath(config_file))
 
     proxy = get_config('proxy') or 'DIRECT'
     proxy_list = proxy.strip('; ') .split(';')

--- a/run.py
+++ b/run.py
@@ -165,16 +165,17 @@ def main():
             format='%(asctime)s <%(module)s.%(funcName)s> %(lineno)d@%(pathname)s \n[%(levelname)s] %(message)s')
         print("DDNS[", __version__, "] run:", os_name, sys.platform)
         print("Configuration was loaded from <==", path.abspath(config_file))
+        print("=" * 25, ctime(), "=" * 25, sep=' ')
 
     proxy = get_config('proxy') or 'DIRECT'
     proxy_list = proxy.strip('; ') .split(';')
 
     cache = get_config('cache', True) and Cache(CACHE_FILE)
     if cache is False:
-        warning("Cache is disabled!")
+        info("Cache is disabled!")
     elif len(cache) < 1 or get_config.time >= cache.time:
+        warning("Cache file is out of dated.")
         cache.clear()
-        print("=" * 25, ctime(), "=" * 25, sep=' ')
     update_ip('4', cache, dns, proxy_list)
     update_ip('6', cache, dns, proxy_list)
 

--- a/run.py
+++ b/run.py
@@ -28,6 +28,8 @@ ddns[%s]
 Copyright (c) New Future (MIT License)
 """ % (__version__)
 
+environ["DDNS_VERSION"] = "${BUILD_SOURCEBRANCHNAME}"
+
 if getattr(sys, 'frozen', False):
     # https://github.com/pyinstaller/pyinstaller/wiki/Recipe-OpenSSL-Certificate
     environ['SSL_CERT_FILE'] = path.join(

